src/wallet/rpcwallet.cpp

```diff
@@ -3366,9 +3408,12 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
     }
 
     if (boost::get<libzcash::SproutPaymentAddress>(&zaddr) != nullptr) {
-        for (SproutNoteEntry & entry : sproutEntries) {
+        auto paginatedSproutEntries = getPaginatedEntries<SproutNoteEntry>(sproutEntries, offset, limit);
```

Comment:
here and in analogical block below for sapling you would call a function returning iterator instead of vector. The for loop below would need to be replaced by `while` statement (to consume the iterator).